File: src/Entity/Categorieoffre.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Repository\CategorieoffreRepository;

#[ORM\Entity(repositoryClass: CategorieoffreRepository::class)]
#[ORM\Table(name: "Categorieoffre")]
#[ORM\Index(name: "typeoffre", columns: ["typeoffre"])]

class Categorieoffre
{
    
#[ORM\Id]
#[ORM\GeneratedValue]
#[ORM\Column]
private ?int  $idcatoffre  = null;

#[ORM\Column]
private ?float $poidsoffre = null;
   

    #[ORM\Column]

private ?int $nbrecolisoffre = null;

#[ORM\Column(name:"typeoffre",unique:true)]  
private ?string  $typeoffre = null;

    public function getIdcatoffre(): ?int
    {
        return $this->idcatoffre ;
    }

    public function getPoidsoffre(): ?float
    {
        return $this->poidsoffre;
    }

    public function setPoidsoffre(float $poidsoffre): self
    {
        $this->poidsoffre = $poidsoffre;

        return $this;
    }

    public function getNbrecolisoffre(): ?int
    {
        return $this->nbrecolisoffre;
    }

    public function setNbrecolisoffre(int $nbrecolisoffre): self
    {
        $this->nbrecolisoffre = $nbrecolisoffre;

        return $this;
    }

    public function getTypeoffre(): ?string
    {
        return $this->typeoffre;
    }

    public function setTypeoffre(string $typeoffre): self
    {
        $this->typeoffre = $typeoffre;

        return $this;
    }

    public function __toString()
    {
        return $this->getTypeoffre(); // Return the title of the task as the string representation
    }
}

File: src/com/mycompany/myapp/entities/Categorieoffre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author utilisateur
 */
public class Categorieoffre {
      private int IdCatOffre,nbreColisOffre ;
     private float poidsOffre;
     private String TypeOffre;

    public Categorieoffre() {
    }

    public Categorieoffre(int IdCatOffre, int nbreColisOffre, float poidsOffre, String TypeOffre) {
        this.IdCatOffre = IdCatOffre;
        this.nbreColisOffre = nbreColisOffre;
        this.poidsOffre = poidsOffre;
        this.TypeOffre = TypeOffre;
    }

    public Categorieoffre(int IdCatOffre, int nbreColisOffre, float poidsOffre) {
        this.IdCatOffre = IdCatOffre;
        this.nbreColisOffre = nbreColisOffre;
        this.poidsOffre = poidsOffre;
        this.TypeOffre = TypeOffre;
    }

    public void setIdCatOffre(int IdCatOffre) {
        this.IdCatOffre = IdCatOffre;
    }

    public void setNbreColisOffre(int nbreColisOffre) {
        this.nbreColisOffre = nbreColisOffre;
    }

    public void setPoidsOffre(float poidsOffre) {
        this.poidsOffre = poidsOffre;
    }

    public void setTypeOffre(String TypeOffre) {
        this.TypeOffre = TypeOffre;
    }

    public Categorieoffre(int nbreColisOffre, float poidsOffre, String TypeOffre) {
        this.nbreColisOffre = nbreColisOffre;
        this.poidsOffre = poidsOffre;
        this.TypeOffre = TypeOffre;
    }

    public int getIdCatOffre() {
        return IdCatOffre;
    }

    public int getNbreColisOffre() {
        return nbreColisOffre;
    }

    public float getPoidsOffre() {
        return poidsOffre;
    }

    public String getTypeOffre() {
        return TypeOffre;
    }

}
